File: An2/MPP/mpp-proiect-csharp-DeepSmeag/repo/NumeProbaRepo.cs
using mpp_proiect_csharp_DeepSmeag.domain;
using mpp_proiect_csharp_DeepSmeag.repo.interfaces;
using mpp_proiect_csharp_DeepSmeag.repo.utils;
using Npgsql;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace mpp_proiect_csharp_DeepSmeag.repo
{
    internal class NumeProbaRepo : INumeProbaRepoInterface
    {
        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
        public bool deleteEntity(int entityId)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "DELETE FROM \"NumeProba\" WHERE \"id\"=@id";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("id", entityId);
                sqlCommand.Prepare();
                sqlCommand.ExecuteNonQuery();
                con.Close();
                return true;
            }
            catch (Exception e)
            {
                return false;
            }
        }

        public List<NumeProba>? getAll()
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "SELECT * FROM \"NumeProba\"";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                using var reader = sqlCommand.ExecuteReader();
                var numeProbe = new List<NumeProba>();
                while (reader.Read())
                {
                    var numeProba = new NumeProba(reader.GetInt32(0), reader.GetString(1));
                    numeProbe.Add(numeProba);
                }
                con.Close();
                return numeProbe;
            }
            catch (Exception e)
            {
                return null;
            }
        }

        public NumeProba? getNumeProbaByName(string name)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "SELECT * FROM \"NumeProba\" WHERE \"nume\"=@nume";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("nume", name);
                sqlCommand.Prepare();
                using var reader = sqlCommand.ExecuteReader();
                if (reader.Read())
                {
                    var numeProba = new NumeProba(reader.GetInt32(0), reader.GetString(1));
                    con.Close();
                    return numeProba;
                }
                return null;
            }
            catch (Exception e)
            {
                return null;
            }
        }

        public NumeProba? getOne(int entityId)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "SELECT * FROM \"NumeProbe\" WHERE \"id\"=@id";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("id", entityId);
                sqlCommand.Prepare();
                using var reader = sqlCommand.ExecuteReader();
                if (reader.Read())
                {
                    var numeProba = new NumeProba(reader.GetInt32(0), reader.GetString(1));
                    con.Close();
                    return numeProba;
                }
                log.Info("Am ajuns dar nu am gasit dupa id");
                con.Close();
                return null;
            }
            catch (Exception e)
            {
                log.Info("Am ajuns la eroare la nume proba");
                log.Error(e.Message);
                return null;
            }
        }

        public bool saveEntity(NumeProba entityType)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "INSERT INTO \"NumeProba\" (\"nume\") VALUES (@nume)";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("nume", entityType.Nume);
                sqlCommand.Prepare();
                sqlCommand.ExecuteNonQuery();
                con.Close();
                return true;
            }
            catch (Exception e)
            {
                return false;
            }
        }

        public bool updateEntity(NumeProba entityType)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "UPDATE \"NumeProba\" SET \"nume\"=@nume WHERE \"id\"=@id";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("nume", entityType.Nume);
                sqlCommand.Parameters.AddWithValue("id", entityType.Id);
                sqlCommand.Prepare();
                sqlCommand.ExecuteNonQuery();
                con.Close();
                return true;
            }
            catch (Exception e)
            {
                return false;
            }
        }
    }
}

File: An2/MPP/mpp-proiect-java-DeepSmeag/persistenceModule/src/main/java/contests/persistence/NumeProbaRepo.java
package contests.persistence;

import contests.model.NumeProba;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import contests.persistence.interfaces.NumeProbaInterface;
import contests.persistence.utils.DBConnection;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class NumeProbaRepo implements NumeProbaInterface {
    private static final Logger logger = LogManager.getLogger();
    @Override
    public List<NumeProba> getAll() {
        try(Connection connection = DBConnection.getNewConnection();
            PreparedStatement ps = connection.prepareStatement("SELECT * FROM \"NumeProbe\"")) {
            ResultSet resultSet = ps.executeQuery();
            ArrayList<NumeProba> numeProbe = new ArrayList<>();
            while (resultSet.next()) {
                Integer id = resultSet.getInt("id");
                String nume = resultSet.getString("nume");
                NumeProba numeProba = new NumeProba(id, nume);
                numeProbe.add(numeProba);
            }
            return numeProbe;
        } catch (SQLException e) {
            return null;
        }
    }

    @Override
    public NumeProba getOne(Integer id) {
        try(Connection connection = DBConnection.getNewConnection();
            PreparedStatement ps = connection.prepareStatement("SELECT * FROM \"NumeProbe\" WHERE id=" + String.valueOf(id) + ";")) {
//            ps.setInt(0, id.intValue());
            ResultSet resultSet = ps.executeQuery();

            resultSet.next();
            String nume = resultSet.getString("nume");
            NumeProba numeProba = new NumeProba(id, nume);

            return numeProba;
        } catch (SQLException e) {
            logger.error(e);
            return null;
        }
    }

    @Override
    public boolean saveEntity(NumeProba entity) {
        try(Connection connection = DBConnection.getNewConnection();
            PreparedStatement ps = connection.prepareStatement("INSERT INTO \"NumeProbe\" (nume) VALUES (?)")) {
            ps.setString(0, entity.getNumeProba());
            ps.executeUpdate();
            return true;
        } catch (SQLException e) {
            return false;
        }
    }

    @Override
    public boolean deleteEntity(Integer id) {
        try(Connection connection = DBConnection.getNewConnection();
            PreparedStatement ps = connection.prepareStatement("DELETE FROM \"NumeProbe\" WHERE id=?")) {
            ps.setInt(0, id);
            ps.executeUpdate();
            return true;
        } catch (SQLException e) {
            return false;
        }
    }

    @Override
    public boolean updateEntity(NumeProba entity) {
        try(Connection connection = DBConnection.getNewConnection();
            PreparedStatement ps = connection.prepareStatement("UPDATE \"NumeProbe\" SET nume=? WHERE id=?")) {
            ps.setString(0, entity.getNumeProba());
            ps.setInt(1, entity.getId());
            ps.executeUpdate();
            return true;
        } catch (SQLException e) {
            return false;
        }
    }

    @Override
    public NumeProba getNumeProbaByName(String name) {
        try(Connection connection = DBConnection.getNewConnection();
            PreparedStatement ps = connection.prepareStatement("SELECT * FROM \"NumeProbe\" WHERE nume=?")) {
            ps.setString(0, name);
            ResultSet resultSet = ps.executeQuery();

            resultSet.next();
            Integer id = resultSet.getInt("id");
            NumeProba numeProba = new NumeProba(id, name);

            return numeProba;
        } catch (SQLException e) {
            return null;
        }
    }
}
